sdk/core/System.Memory.Data/tests/BinaryDataTests.cs

```diff
@@ -299,6 +299,8 @@ public void CanCreateBinaryDataFromCustomType()
             AssertData(BinaryData.FromObjectAsJson(payload));
             AssertData(BinaryData.FromObjectAsJson(payload, new Text.Json.JsonSerializerOptions { IgnoreNullValues = true }));
             AssertData(new BinaryData(payload, type: typeof(TestModel)));
+            AssertData(new BinaryData(payload));
```

Comment:
We should have a test that passes instance of a subtype but Type representing the instance's base type. The test should verify that only the base properties are serialized.